src/components/Modal/Modal.tsx

```diff
@@ -1,34 +1,27 @@
 import { Dialog as BaseDialog } from '@kobalte/core';
-import { ComponentProps, ParentComponent, splitProps } from 'solid-js';
+import type { Component, ComponentProps } from 'solid-js';
 
-import { buttonRecipe, ButtonVariants } from '@cubeartisan/cubeartisan/components/Button';
 import * as styles from '@cubeartisan/cubeartisan/components/Modal/Modal.css';
 
-const ModalTrigger: ParentComponent<{ recipe?: ButtonVariants } & ComponentProps<typeof BaseDialog.Trigger>> = (
-  props,
-) => {
-  const [local, others] = splitProps(props, ['recipe']);
+const ModalTrigger = BaseDialog.Trigger;
 
-  return <BaseDialog.Trigger class={buttonRecipe(local.recipe)} {...others} />;
-};
-
-const ModalCloseButton: ParentComponent<{ recipe?: ButtonVariants } & ComponentProps<typeof BaseDialog.CloseButton>> = (
-  props,
-) => {
-  const [local, others] = splitProps(props, ['recipe']);
-
-  return <BaseDialog.CloseButton class={buttonRecipe(local.recipe)} {...others} />;
-};
+const ModalCloseButton = BaseDialog.CloseButton;
 
 const ModalPortal = BaseDialog.Portal;
 
-const ModalOverlay: typeof BaseDialog.Overlay = (props) => <BaseDialog.Overlay class={styles.overlay} {...props} />;
+const ModalOverlay: Component<ComponentProps<typeof BaseDialog.Overlay>> = (props) => (
```

Comment:
How is this different from `typeof  BaseDialog.Overlay`?

Reply:
It correctly types the props. Before this change, props was being read by typescript as implicit any